backend/server/app.py

```
@@ -210,3 +261,296 @@ async def health_check():
+                                y_coords.append(float(dte))
+                                z_coords.append(float(vol))
+        
+        return {
+            "x": x_coords,
+            "y": y_coords,
+            "z": z_coords,
+            "timestamp": datetime.now().isoformat()
+        }
+    except Exception as e:
+        logging.error(f"Error getting surface snapshot: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Error getting surface snapshot: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)